src/sync/sync_manager.cpp

```diff
@@ -32,6 +35,73 @@ SyncManager& SyncManager::shared()
     return manager;
 }
 
+void SyncManager::configure_file_system(const std::string& base_file_path,
+                                        MetadataMode metadata_mode,
+                                        util::Optional<std::vector<char>> custom_encryption_key)
+{
+    std::vector<std::tuple<std::string, std::shared_ptr<SyncUser>>> users_to_add;
+    REALM_ASSERT(!m_file_manager && !m_metadata_manager);   // Can only call this once
+    m_file_manager = std::make_unique<SyncFileManager>(base_file_path);
+    switch (metadata_mode) {
+        case MetadataMode::NoEncryption:
+            m_metadata_manager = std::make_unique<SyncMetadataManager>(m_file_manager->metadata_path(),
+                                                                       false);
+            break;
+        case MetadataMode::Encryption:
+            m_metadata_manager = std::make_unique<SyncMetadataManager>(m_file_manager->metadata_path(),
+                                                                       true,
+                                                                       std::move(custom_encryption_key));
+            break;
+        case MetadataMode::NoMetadata:
+            return;
+    }
+
+    REALM_ASSERT(m_metadata_manager);
+    // Load persisted users into the users map.
+    SyncUserMetadataResults users = m_metadata_manager->all_unmarked_users();
+    for (size_t i = 0; i < users.size(); i++) {
+        // Note that 'admin' style users are not persisted.
+        auto user_data = users.get(i);
+        auto user_token = user_data.user_token();
+        auto identity = user_data.identity();
+        if (user_token) {
+            users_to_add.emplace_back(std::make_tuple(identity,
+                                                      std::make_shared<SyncUser>(*user_token,
+                                                                                 std::move(identity),
+                                                                                 false)));
+        }
+    }
+    // Delete any users marked for death.
+    std::vector<SyncUserMetadata> dead_users;
+    SyncUserMetadataResults users_to_remove = m_metadata_manager->all_users_marked_for_removal();
+    for (size_t i = 0; i < users_to_remove.size(); i++) {
```

Comment:
This type of collection doesn't support foreach iteration. I hope to add support in a followup item, and will change the loops when that happens.